comp/otelcol/provider/def/component.go

```diff
@@ -0,0 +1,25 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+package provider
+
+import (
+	"go.opentelemetry.io/collector/otelcol"
+)
+
+// team: opentelemetry
+
+// Component is the component type.
+type Component interface {
+	ExtendedConfigProvider
```

Comment:
Why have two interfaces instead of just moving everything into this one?

Reply:
Moved everything into the Component interface 👍